core/src/Kokkos_Array.hpp

```diff
@@ -118,6 +118,26 @@ struct Array {
     return &m_internal_implementation_private_member_data[0];
   }
 
+  KOKKOS_INLINE_FUNCTION constexpr pointer begin() noexcept { return data(); }
+
+  KOKKOS_INLINE_FUNCTION constexpr const_pointer begin() const noexcept {
```

Comment:
`data()` is not marked `noexcept` and probably should be, both to be consistent with `std::array` and a `noexcept` function like `end()` should call other `noexcept` functions.

That being said, I can't imagine how it will ever matter in practice...